applications/adv_networking_bench/cpp/rdma_bench.h

```diff
@@ -0,0 +1,185 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "holoscan/holoscan.hpp"
+#include <queue>
+#include <arpa/inet.h>
+#include <assert.h>
+#include <sys/time.h>
+
+namespace holoscan::ops {
+
+class AdvNetworkingRdmaOp : public Operator {
+ public:
+  HOLOSCAN_OPERATOR_FORWARD_ARGS(AdvNetworkingRdmaOp)
+
+  AdvNetworkingRdmaOp() = default;
+
+  ~AdvNetworkingRdmaOp() {
+    HOLOSCAN_LOG_INFO(
+        "Finished receiver with {}/{} bytes/packets received and {}/{} bytes/packets sent",
+        ttl_bytes_recv_,
+        ttl_pkts_recv_,
+        ttl_bytes_sent_,
+        ttl_pkts_sent_);
+
+    HOLOSCAN_LOG_INFO("ANO benchmark clent op shutting down");
+    freeResources();
+  }
+
+  void initialize() override {
+    HOLOSCAN_LOG_INFO("AdvNetworkingRdmaOp::initialize()");
+    holoscan::Operator::initialize();
+
+    if (send_.get()) {
+      send_mr_name_ = server_.get() ? "DATA_TX_CPU_SERVER" : "DATA_TX_CPU_CLIENT";
+    }
+    if (receive_.get()) {
+      receive_mr_name_ = server_.get() ? "DATA_RX_CPU_SERVER" : "DATA_RX_CPU_CLIENT";
+    }
+
+    HOLOSCAN_LOG_INFO("AdvNetworkingRdmaOp::initialize() complete in {} mode",
+                      server_.get() ? "server" : "client");
+  }
+
+  void freeResources() {
+    HOLOSCAN_LOG_INFO("AdvNetworkingRdmaOp::freeResources() start");
+    HOLOSCAN_LOG_INFO("AdvNetworkingRdmaOp::freeResources() complete");
+  }
+
+  void setup(OperatorSpec& spec) override {
+    spec.param<int>(message_size_, "message_size", "Message size", "Message size in bytes", 1024);
+    spec.param<std::string>(
+        server_addr_str_, "server_address", "Server address", "Server address", "192.168.3.1");
+    spec.param<std::string>(
+        client_addr_str_, "client_address", "Client address", "Client address", "192.168.2.1");
```

Comment:
can we call these `IP` ? address is used a lot and can be mixed with pcie bus address, ethernet address.